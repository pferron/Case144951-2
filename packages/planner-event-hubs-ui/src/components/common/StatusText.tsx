import React from 'react';
import { StatusLabel } from '@cvent/carina/components/StatusLabel';
import { useTranslate } from 'nucleus-text';

interface Props {
  isActive: boolean;
  activeLabelTextKey?: string;
  inactiveLabelTextKey?: string;
}
function StatusText({
  isActive,
  activeLabelTextKey = 'channel_status_active',
  inactiveLabelTextKey = 'channel_status_inactive'
}: Props): JSX.Element {
  const { translate } = useTranslate();
  return isActive ? (
    <StatusLabel variant="positive">{translate(activeLabelTextKey)}</StatusLabel>
  ) : (
    <StatusLabel variant="neutral">{translate(inactiveLabelTextKey)}</StatusLabel>
  );
}
export default StatusText;
