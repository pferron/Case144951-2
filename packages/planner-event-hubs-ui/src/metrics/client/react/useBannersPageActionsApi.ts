/**
 * ------------------------------------------------------------------------------------
 * This client was automatically generated. ** Do Not Edit **
 * ------------------------------------------------------------------------------------
 */
import { useCallback } from 'react';
import * as Rudder from '../rudder-web';
import { usePageActions } from '@cvent/sli-nextjs-metrics';
import { ApiCallback } from '@rudderstack/analytics-js';
import { useIdentityApi } from '../IdentityProvider';
import { TrackProperties } from '../commonTypes';

export type ActionDefinition = {
  id: string;
  destinations?: string[];
};

export interface AssignPagesButtonClicked extends TrackProperties {}
export interface BannerContentSectionChecked extends TrackProperties {
  // Title provided for the banner
  bannerTitle: string;
  // Description provided for the banner
  bannerBody?: string;
  // Button text if there are any buttons added to the banner
  buttonText?: string;
  // Destination page or link on click on the button
  whereWillThisButtonBringYourMembers?: string;
  // Name of the page that button takes when selected Events+ hub
  selectAPage?: string;
  // External Link to which the button clicks takes to
  externalLink?: string;
}
export interface BannerDesignSectionChecked extends TrackProperties {
  // Text alignment selected for the banner
  textAlignment: string;
  // Font color of the text that appears on the banner
  fontColor?: string;
}
export interface BannerImageChecked extends TrackProperties {}
export interface BannerNameProvided extends TrackProperties {
  // Template chosen for the banner
  bannerTemplate: string;
  // Name provided for the banner
  bannerName: string;
}
export interface CreateBannerButtonClicked extends TrackProperties {}
export interface PagesAssignButtonClicked extends TrackProperties {
  // Name of the page assigned to the banner
  bannerPlacement: string;
}

interface AnalyticsApi {
  assignPagesButtonClicked: (
    props?: AssignPagesButtonClicked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerContentSectionChecked: (
    props: BannerContentSectionChecked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerDesignSectionChecked: (
    props: BannerDesignSectionChecked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerImageChecked: (props?: BannerImageChecked, options?: Rudder.Options, callback?: ApiCallback) => void;
  bannerNameProvided: (props: BannerNameProvided, options?: Rudder.Options, callback?: ApiCallback) => void;
  createBannerButtonClicked: (
    props?: CreateBannerButtonClicked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  pagesAssignButtonClicked: (props: PagesAssignButtonClicked, options?: Rudder.Options, callback?: ApiCallback) => void;
}

/**
 * This hook exposes a typed api with functions from the application's tracking plan.
 * Internally, each function uses the usePageActions API to track actions and forward
 * them to the reporters that were installed during the Metrics initialization in your
 * application's main entry point component.
 *
 * Note: This is the recommended way of tracking/measuring actions as it allows you to
 * register base properties during the metrics' client initialization, which will be included
 * automatically to every action whenever they are executed by your instrumentation code
 *
 *
 * How to use it:
 *
 * `const { acqSignupCompleted } = useBannersPageActionsApi();`
 *
 *  // Somewhere in your application you can use this function to track this action
 * `acqSignupCompleted({ userId, email, accountName });`
 *
 * // If you want to track this action as an error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, error: '500 Error........' });`
 *
 * // If you want to track this action as an expected error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, expected_error: 'This email address is already associated with an existing account.' });`
 *
 * @returns AnalyticsApi
 */
export const useBannersPageActionsApi = (): AnalyticsApi => {
  const { trackAction, getBaseProps } = usePageActions();
  const { trackingPlanName } = getBaseProps() || {};
  const { updateProfile } = useIdentityApi();

  /**
   * this action tracks the users who has intent to assign pages to banners in order to show banners on those pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : assign pages button clicked
   * downstreamName: assign pages button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on 'Assign pages' in page section of banner
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {AssignPagesButtonClicked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const assignPagesButtonClicked = useCallback(
    (props?: AssignPagesButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'assign pages button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'assign pages button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who provides banner details in order to update the banner content section
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner content section checked
   * downstreamName: banner content section checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of banner content section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerContentSectionChecked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerContentSectionChecked = useCallback(
    (props: BannerContentSectionChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner content section checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner content section checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who updates design section in order to change the text alignments in banner
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner design section checked
   * downstreamName: banner design section checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of banner design section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerDesignSectionChecked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerDesignSectionChecked = useCallback(
    (props: BannerDesignSectionChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner design section checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner design section checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who saves the image in order to use in banners
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner image checked
   * downstreamName: banner image checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of website image section in banners section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerImageChecked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerImageChecked = useCallback(
    (props?: BannerImageChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner image checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner image checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who provide a name to banners in order to create the banner
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner name provided
   * downstreamName: banner name provided
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerNameProvided} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerNameProvided = useCallback(
    (props: BannerNameProvided, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner name provided',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner name provided'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who wants to create banner in order to use in eventsplus pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : create banner button clicked
   * downstreamName: create banner button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when the user clicks on Create Banner' button on the banners page
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {CreateBannerButtonClicked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const createBannerButtonClicked = useCallback(
    (props?: CreateBannerButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'create banner button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'create banner button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who assigns the pages to banners in order to show the banner on those pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : pages assign button clicked
   * downstreamName: pages assign button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on 'Assign' in the select 'Pages' modal
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {PagesAssignButtonClicked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const pagesAssignButtonClicked = useCallback(
    (props: PagesAssignButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'pages assign button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'pages assign button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );

  return {
    assignPagesButtonClicked,
    bannerContentSectionChecked,
    bannerDesignSectionChecked,
    bannerImageChecked,
    bannerNameProvided,
    createBannerButtonClicked,
    pagesAssignButtonClicked
  };
};
