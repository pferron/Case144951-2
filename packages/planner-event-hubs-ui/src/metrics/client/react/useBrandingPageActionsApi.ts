/**
 * ------------------------------------------------------------------------------------
 * This client was automatically generated. ** Do Not Edit **
 * ------------------------------------------------------------------------------------
 */
import { useCallback } from 'react';
import * as Rudder from '../rudder-web';
import { usePageActions } from '@cvent/sli-nextjs-metrics';
import { ApiCallback } from '@rudderstack/analytics-js';
import { useIdentityApi } from '../IdentityProvider';
import { TrackProperties } from '../commonTypes';

export type ActionDefinition = {
  id: string;
  destinations?: string[];
};

export interface BrandingSaveButtonClicked extends TrackProperties {
  // Primary color of the events+ hub
  primaryColor: string;
  // Secondary color color of the events+ hub
  secondaryColor: string;
  // BackgroundColor color color of the events+ hub
  backgroundColor: string;
  // Mood of the events+ hub
  selectYourMood: string;
  // Whether safemode is turned on or off for the events+ hub
  safeColorMode: boolean;
  // Whether background color is overriden for the events+ hub
  overrideBackgroundColor?: boolean;
}
export interface LogoImageChecked extends TrackProperties {}

interface AnalyticsApi {
  brandingSaveButtonClicked: (
    props: BrandingSaveButtonClicked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  logoImageChecked: (props?: LogoImageChecked, options?: Rudder.Options, callback?: ApiCallback) => void;
}

/**
 * This hook exposes a typed api with functions from the application's tracking plan.
 * Internally, each function uses the usePageActions API to track actions and forward
 * them to the reporters that were installed during the Metrics initialization in your
 * application's main entry point component.
 *
 * Note: This is the recommended way of tracking/measuring actions as it allows you to
 * register base properties during the metrics' client initialization, which will be included
 * automatically to every action whenever they are executed by your instrumentation code
 *
 *
 * How to use it:
 *
 * `const { acqSignupCompleted } = useBrandingPageActionsApi();`
 *
 *  // Somewhere in your application you can use this function to track this action
 * `acqSignupCompleted({ userId, email, accountName });`
 *
 * // If you want to track this action as an error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, error: '500 Error........' });`
 *
 * // If you want to track this action as an expected error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, expected_error: 'This email address is already associated with an existing account.' });`
 *
 * @returns AnalyticsApi
 */
export const useBrandingPageActionsApi = (): AnalyticsApi => {
  const { trackAction, getBaseProps } = usePageActions();
  const { trackingPlanName } = getBaseProps() || {};
  const { updateProfile } = useIdentityApi();

  /**
   * this action tracks the users who makes changes in themes in order to change the colors of branding
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : branding save button clicked
   * downstreamName: branding save button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when clicks on save button in brandind edit page
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BrandingSaveButtonClicked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const brandingSaveButtonClicked = useCallback(
    (props: BrandingSaveButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'branding save button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'branding save button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who saves the image in order to use as logo in events+
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : logo image checked
   * downstreamName: logo image checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of website image section in branding section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {LogoImageChecked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const logoImageChecked = useCallback(
    (props?: LogoImageChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'logo image checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'logo image checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );

  return {
    brandingSaveButtonClicked,
    logoImageChecked
  };
};
