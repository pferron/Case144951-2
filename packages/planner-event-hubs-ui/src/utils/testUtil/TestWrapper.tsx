import React, { useMemo } from 'react';
import { LocalizationProvider } from 'nucleus-text';
import { Breakpoints } from '@cvent/carina/components/Breakpoints';
import { AppFeature } from '@cvent/planner-event-hubs-model/types';
import { ThemeProvider } from '@cvent/carina/components/ThemeProvider/ThemeProvider';
import { getDefaultTheme } from '@cvent/carina/utils/tokens';
import { AppFeaturesProvider } from '@components/AppFeaturesProvider';
import { UserPermissionsContext } from '@hooks/UserPermissionsProvider';
import locales, { defaultLocale } from '../../../locales';

export function TestWrapper({ children, appFeatures }: Props): JSX.Element {
  const userPermissionsValue = useMemo(() => ({ VideoCenter: true }), []);
  return (
    <ThemeProvider theme={getDefaultTheme()}>
      <LocalizationProvider locale={defaultLocale} locales={locales}>
        <Breakpoints ssrSize="m" key={typeof window === 'undefined' ? 'ssr' : 'csr'}>
          <AppFeaturesProvider value={appFeatures}>
            <UserPermissionsContext.Provider value={userPermissionsValue}>{children}</UserPermissionsContext.Provider>
          </AppFeaturesProvider>
        </Breakpoints>
      </LocalizationProvider>
    </ThemeProvider>
  );
}

interface Props {
  children: JSX.Element;
  appFeatures?: AppFeature[];
}
