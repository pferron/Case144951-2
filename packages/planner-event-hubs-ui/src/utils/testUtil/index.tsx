import { render, configure } from '@testing-library/react';
import { TestWrapper } from './TestWrapper';

configure({ testIdAttribute: 'data-cvent-id' });

type RenderParams = Parameters<typeof render>;
type RenderReturn = ReturnType<typeof render>;
const customRender = (ui: RenderParams[0], options?: RenderParams[1]): RenderReturn =>
  render(ui, { wrapper: TestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
