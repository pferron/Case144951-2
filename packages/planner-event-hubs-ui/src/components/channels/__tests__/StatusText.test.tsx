import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import StatusText from '@components/common/StatusText';
import { TestWrapper } from '@utils/testUtil/TestWrapper';
import 'jest-axe/extend-expect';

describe('Status Text', () => {
  it('Render Active label', async () => {
    const { baseElement } = render(
      <TestWrapper>
        <StatusText isActive />
      </TestWrapper>
    );
    expect(baseElement).toMatchSnapshot();
    await new Promise(resolve => {
      setTimeout(resolve, 0);
    });
    await waitFor(() => {
      expect(screen.getByText('Live')).toBeInTheDocument();
    });
    // HUB-137682
    // expect(await axe(baseElement)).toHaveNoViolations();
  });

  it('Render Inactive label', async () => {
    const { baseElement } = render(
      <TestWrapper>
        <StatusText isActive={false} />
      </TestWrapper>
    );
    expect(baseElement).toMatchSnapshot();
    await new Promise(resolve => {
      setTimeout(resolve, 0);
    });
    await waitFor(() => {
      expect(screen.getByText('Not live')).toBeInTheDocument();
    });
    // HUB-137682
    // expect(await axe(baseElement)).toHaveNoViolations();
  });
});
