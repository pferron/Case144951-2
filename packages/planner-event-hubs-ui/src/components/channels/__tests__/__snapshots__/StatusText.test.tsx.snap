// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Text Render Active label 1`] = `
.emotion-0 {
  border-width: 0;
  border-radius: 4px;
  padding-right: 8px;
  padding-left: 8px;
  padding-top: 0;
  padding-bottom: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  max-width: 100%;
  box-sizing: border-box;
  background-color: #D1EAD1;
}

.emotion-0 svg {
  fill: rgb(26,32,38);
}

.emotion-1 {
  font-family: 'Rubik','Helvetica','Arial';
  font-size: 0.75rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  vertical-align: baseline;
  box-sizing: border-box;
  border: 0;
  padding: 5px 0;
  position: relative;
  color: #006200;
}

<body>
  <div>
    <span
      class="emotion-0"
    >
      <span
        class="emotion-1"
      >
        Live
      </span>
    </span>
  </div>
</body>
`;

exports[`Status Text Render Inactive label 1`] = `
.emotion-0 {
  border-width: 0;
  border-radius: 4px;
  padding-right: 8px;
  padding-left: 8px;
  padding-top: 0;
  padding-bottom: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  max-width: 100%;
  box-sizing: border-box;
  background-color: #D3E5FA;
}

.emotion-0 svg {
  fill: rgb(26,32,38);
}

.emotion-1 {
  font-family: 'Rubik','Helvetica','Arial';
  font-size: 0.75rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  vertical-align: baseline;
  box-sizing: border-box;
  border: 0;
  padding: 5px 0;
  position: relative;
  color: #0050AB;
}

<body>
  <div>
    <span
      class="emotion-0"
    >
      <span
        class="emotion-1"
      >
        Not live
      </span>
    </span>
  </div>
</body>
`;
